pkg/engine/turn/turn.go

```diff
@@ -123,67 +140,76 @@ func (mgr *manager) RemoveTarget(id key.TargetID) {
+//	4. loop through all targets in the order, subtracing this AV from them (active target gauge = 0)
+//			new_gauge = current_gauge - av * speed
 func (mgr *manager) StartTurn() (key.TargetID, float64, []event.TurnStatus, error) {
 	if mgr.activeTurn {
 		return -1, 0, nil, fmt.Errorf("cannot start turn when already in an active turn: %+v", mgr)
 	}
 
-	// reset gauge cost for this new turn
 	mgr.gaugeCost = 1.0
 	mgr.activeTurn = true
 	mgr.activeTarget = mgr.order.order[0].id
 	av := mgr.av(mgr.activeTarget)
 
-	// mgr.order.order[0].gauge = 0 // set gauge/av of active to 0
-
 	for _, t := range mgr.order.order {
 		t.gauge -= av * mgr.attr.Stats(t.id).SPD()
 	}
 
-	// TODO: Set the active turn to the target at the top of the order (target with lowest AV).
-	//		Get the current AV of this target and subtract it from all targets in the order to simulate
-	//		"progressing time forward" by the given AV.
-	//
-	//	1. Mark target at top of order as "active"
-	//	2. get that target's current AV
-	//	3. add AV to "TotalAV" (keeps track of how much AV has progressed over entire sim)
-	//	4. loop through all targets in the order, subtracing this AV from them (active target gauge = 0)
-	//			new_gauge = current_gauge - av * speed
-
 	mgr.totalAV += av
-	return mgr.activeTarget, av, nil, nil
+	return mgr.activeTarget, av, mgr.EventTurnStatus(), nil
 }
 
+// ResetTurn resets the gauge of the target taking their turn (target at top of stack) and updates the Manager's turnOrder.
+// New gauge is set at BaseGauge * gaugeCost
+// Target should then be moved to the bottom of the turn order and then sorted up to the correct
+// position based on their AV. In the event of ties, this target should be last in the order.
+// Note that this is the same behavior as AddTarget (but different from SetGauge which goes top-down).
+// 1. update gauge of active target
+// 2. move target to bottom of order
+// 3. sort target up based on AV. In the event of tie, this target should be at bottom of order.
+// 4. update targetIndexes for all targets that moved in the order (or just repopulate all)
+// 5. emit TurnResetEvent
 func (mgr *manager) ResetTurn() error {
 	if !mgr.activeTurn {
 		return fmt.Errorf(
 			"target at top of order must have 0 gauge to call reset (their turn is active) %+v", mgr.order.order[0])
 	}
+
 	mgr.activeTurn = false
 	mgr.order.order[0].gauge = BaseGauge * mgr.gaugeCost
+	
+	// It would be more efficient to loop through mgr.order ourselves to determine this single target's placement instead of resorting the whole array when no other elements are changing.
+	// Unless we are also checking for other SPD changes that happened during the turn, in which case sort.Stable() is better to use, but only after we move the element to the end
+	// so as to ensure that, in the case of a tie, it is properly at the tail end of the tied elements.
+	mgr.order.order = append(mgr.order.order[1:], mgr.order.order[0])
```

Comment:
nit: might want to rename the variables since `mgr.order.order` is not clear what it does

Reply:
Renamed.